modules/HapticGlove_module/src/RobotSkin.cpp

```diff
@@ -360,26 +370,38 @@ bool RobotSkin::getRawTactileFeedbackFromRobot()
 
 void RobotSkin::updateTactileFeedbacks()
 {
+    std::lock_guard<std::mutex> lock(m_mutex);
+
     this->getRawTactileFeedbackFromRobot();
 
     this->updateCalibratedTactileData();
 
-    this->computeAreFingersInContact();
-
     this->computeMaxContactStrength();
 
+    this->computeAreFingersInContact();
+
     this->computeVibrotactileFeedback();
 }
 
 void RobotSkin::computeAreFingersInContact()
 {
     for (size_t i = 0; i < m_noFingers; i++)
     {
-        m_areFingersInContact[i] = m_fingersTactileData[i].maxTactileFeedbackAbsoluteValue()
+        m_areFingersInContact[i] = m_fingersContactStrength[i]
                                    > m_fingersTactileData[i].contactThreshold();
-
-        m_areFingersContactChanges[i] = m_fingersTactileData[i].maxTactileFeedbackDerivativeValue()
-                                        > m_fingersTactileData[i].contactDerivativeThreshold();
+        // if (m_areFingersInContact[i]) {
+        //     yInfo() << m_logPrefix << "Finger in contact " << i;
+        //     yInfo() << "value: " << m_fingersContactStrength[i];
+        //     yInfo() << "treshold: " << m_fingersTactileData[i].contactThreshold();
+        // }
+        m_areFingersContactChanges[i] = m_areFingersInContact[i]
+                                        && (m_fingersContactStrengthDerivate[i]
+                                        > m_fingersTactileData[i].contactDerivativeThreshold());
+        // if (m_areFingersContactChanges[i]) {
+        //     yInfo() << m_logPrefix << ">>> Finger in contact derivative " << i;
+        //     yInfo() << ">>> value: " << m_fingersContactStrengthDerivate[i];
+        //     yInfo() << ">>> treshold: " << m_fingersTactileData[i].contactDerivativeThreshold();
+        // }
```

Comment:
Could you avoid leaving commented code? In case it might still be needed, could you leave a comment?

The same below